### ISSUE TITLE
distsql: panic: merged columns with ON condition

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#20569]
https://sentry.io/cockroach-labs/cockroachdb/issues/415013547/

```
*log.safeError: stopper.go:179: session.go:1366
  File "github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go", line 423, in func1
  File "github.com/cockroachdb/cockroach/pkg/sql/session.go", line 1381, in maybeRecover
  File "github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go", line 1927, in createPlanForJoin
  File "github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go", line 2052, in createPlanForNode
  File "github.com/cockroachdb/cockroach/pkg/sql/distsql_physical_planner.go", line 2055, in createPlanForNode
...
(22 additional frame(s) were not displayed)

stopper.go:179: session.go:1366
```

**Comment by RaduBerinde** [comment]
This is `panic: merged columns with ON condition`. This was fixed in 1.2 in #19660 and subsequent changes. These changes are too extensive to put in 1.1.

I'm hesitant to investigate a one-off fix for 1.1 unless we see this happening more frequently.

**Comment by jordanlewis** [comment]
Is this closed now?
